salt/modules/event.py

```diff
@@ -90,14 +90,16 @@ def fire_master(data, tag, preload=None):
                 # Ensure ret is True.
                 ret = True
             except Exception:
-                ret = False
+                # only set a False ret if it hasn't been sent atleast once
+                if ret is None:
+                    ret = False
         return ret
     else:
         # Usually, we can send the event via the minion, which is faster
         # because it is already authenticated
         try:
-            return salt.utils.event.MinionEvent(__opts__, listen=False).fire_event(
-                {'data': data, 'tag': tag, 'events': None, 'pretag': None}, 'fire_master')
+            me = salt.utils.event.MinionEvent(__opts__, listen=False, keep_loop=True)
+            me.fire_event({'data': data, 'tag': tag, 'events': None, 'pretag': None}, 'fire_master')
```

Comment:
@mattp- you've removed the `return` statement here. Just missed it? If not could you please comment this?